### DISCUSSION TITLE
Breakdown of event prop with filter possible?

### DISCUSSION THREAD

**Discussion by csheppard** [plausible/analytics#913] [Self-Hosted Support] [answered]
Hi

Using v1.3.0-rc.0.

I'm tracking this goal:
```javascript
plausible('Click', { props: { accountId, itemId } });
```

I'd like to get a breakdown of `itemId` via the stats API when filtering with `accountId`

e.g. `/api/v1/stats/breakdown?site_id=mysite.com&period=7d&property=event:props:itemId&filters=event:name==Click;event:props:accountId==123&metrics=visitors`

But this returns no results. Is this kind of query not possible?

Thanks!

**Comment by ukutaht** [comment, marked as answer]
Unfortunately there is a limitation with the Clickhouse `Nested` datastructure that prevents us from doing a breakdown by one property while filtering on another. Sorry, this should be explained in the documentation.

Clickhouse is working on a `Map` data type that will probably work better for custom props and will allow us to fix this problem. It's still unstable though so it will take some time until it's thoroughly tested and ready to be used.

Sorry I don't have a better answer :/

**Reply by csheppard** [reply]
No worries, thank you for explaining.

**Reply by kvn-shn** [reply]
Oh, I'm still evaluating Plausible in the trial period but finding this out is almost a deal breaker (coming from GA). It simply makes tracking multiple properties a lot less useful if they cannot be filtered together. @ukutaht I'm really hoping you're keeping an eye on Clickhouse's [`Map`](https://clickhouse.tech/docs/en/sql-reference/data-types/map/). Thanks.

**Reply by reefki** [reply]
@ukutaht any update regarding `Map` data-type implementation? I noticed it's now no longer an experimental feature: https://clickhouse.com/docs/en/sql-reference/data-types/map

**Reply by ukutaht** [reply]
The issue with Map is that it has to be typed like `Map(String, String)`. But what if you want an integer in the value?

For that they're building a JSON data type that can hold arbitrary data that looks like JSON. I think that's what we want. It came out in a recent Clickhouse version upgrade so that's the new plan.

In terms of capacity, I have no idea when we can tackle this.

**Reply by Funnelll** [reply]
We would love to migrate to Plausible from GA but most of the reports used by our App require us to query multiple parameters/dimensions at the same time. Is there a new update on the ETA for this, please?

Do you plan to have a limit on the parameters/dimensions used by a single query once this new update is implemented, please?

**Reply by justinleebehnke** [reply]
Just in case it helps someone, here could be a workaround, it might be a little bit ugly but it worked for me.

```typescript
plausible('Click', { props: { accountId, itemId, accountIdAndItemId: `${accountId}:${itemId}` } });
```
This new prop will be visible in the Plausible UI and it will allow you to break down by that combination of items and see the number of "Clicks" on that combination of "accountId" and "itemId" rather than just one or the other in summation.

In my case, I was working on some data that I wanted to plot and I was able to split on the `:` delimiter to break my data apart so that I could see the different dimensions before formatting it to work with the FusionChart that I wanted. This was me querying the `/breakdown` API endpoint btw.
